aiida/engine/launch.py

```diff
@@ -118,8 +119,38 @@ def submit(process: TYPE_SUBMIT_PROCESS, **inputs: Any) -> ProcessNode:
 
     # Do not wait for the future's result, because in the case of a single worker this would cock-block itself
     runner.controller.continue_process(process_inited.pid, nowait=False, no_reply=True)
+    node = process_inited.node
+
+    if not wait:
+        return node
+
+    while not node.is_terminated:
+        time.sleep(wait_interval)
+        LOGGER.report(f'Process<{node.pk}> has not yet terminated, current state is `{node.process_state}`.')
+
+    return node
+
+
+def await_processes(nodes: t.Sequence[ProcessNode], sleep_interval: int = 1) -> None:
```

Comment:
Maybe also call it `wait_interval` here, to be consistent with the `submit` argument?